apps/scan/frontend/src/utils/use_voter_settings_controls.ts

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import {
+  VoterSettingsManagerContext,
+  useCurrentLanguage,
+  useLanguageControls,
+  useQueryChangeListener,
+} from '@votingworks/ui';
+import { DefaultTheme, ThemeContext } from 'styled-components';
+import { LanguageCode } from '@votingworks/types';
+import { getAuthStatus, getScannerStatus } from '../api';
+
+interface VoterSettingsControls {
+  resetVoterSettings: () => void;
+}
+
+/**
+ * useVoterSettingsControls adds Scan-specific settings to the base settings provided by VoterSettingsManagerContext
+ */
+export function useVoterSettingsControls(): VoterSettingsControls {
+  const languageContext = useLanguageControls();
+  const voterSettingsContext = React.useContext(VoterSettingsManagerContext);
+  const currentLanguage = useCurrentLanguage();
+  const currentTheme = React.useContext(ThemeContext);
+
+  // Queries for listeners
+  const authStatusQuery = getAuthStatus.useQuery();
+  const scannerStatusQuery = getScannerStatus.useQuery();
+
+  // Voter session specific settings, saved to return to after auth-ed sessions
+  const [voterSessionTheme, setVoterSessionTheme] =
+    React.useState<DefaultTheme | null>(null);
+  const [voterSessionLanguage, setVoterSessionLanguage] =
+    React.useState<LanguageCode | null>(null);
```

Comment:
Maybe rename to `savedVoterSessionTheme` and `savedVoterSessionLanguage` for clarity